src/platform/backends/libvirt/libvirt_virtual_machine.cpp

```diff
@@ -240,25 +243,37 @@ mp::LibVirtVirtualMachine::LibVirtVirtualMachine(const mp::VirtualMachineDescrip
                                                  const std::string& bridge_name, mp::VMStatusMonitor& monitor)
     : VirtualMachine{desc.key_provider, desc.vm_name},
       connection{connection},
-      domain{get_domain_definition(connection, desc, bridge_name)},
-      mac_addr{get_instance_mac_addr(domain.get())},
+      domain{domain_definition_for(connection, desc, bridge_name)},
+      mac_addr{instance_mac_addr_for(domain.get())},
       username{desc.ssh_username},
-      ip{get_instance_ip(connection, mac_addr)},
+      ip{instance_ip_for(connection, mac_addr)},
       monitor{&monitor}
 {
-    state = get_domain_state(domain.get());
+    state = domain_state_for(domain.get());
+}
+
+mp::LibVirtVirtualMachine::~LibVirtVirtualMachine()
+{
+    update_suspend_status = false;
+
+    if (state == State::running)
+        suspend();
```

Comment:
So we suspend on deletion of `LibVirtVirtualMachine`? But only if it's running? Are we sure that's right? And that it shouldn't shut down instead?

Reply:
So, what I'm trying to accomplish here is when the daemon is `SIGTERM`'d with running VM's, like when a `snap refresh` is done, we can at least save the state of the VM and then when the daemon restarts, the VM will be started using the saved image.

If a VM is being deleted, the daemon will explicitly shut it down if it is running before destroy the `LibVirtVirtualMachine` object.